src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -565,9 +565,14 @@ private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
                     return null;
                 }
 
-                if(s_referencedic.ContainsKey(assemblyname))
+                if (s_referencedic.ContainsKey(assemblyname))
```

Comment:
I do not know much about the context when this is executed, but would it be possible that nuget has downloaded and added reference to the project? (or if runtime load is needed then perhaps at least some library nuget perhaps provides)